gamla/graph.py

```diff
@@ -8,29 +8,63 @@
 from gamla import functional, functional_generic
 
 
+@toolz.curry
+def _is_better_node(node: Any, seen: Dict, key: Callable):
+    old_node = seen[key(node)]
+    return node > old_node
+
+
+@toolz.curry
+def _save_to_dict(element: Any, dictionary: Dict, func: Callable):
+    dictionary[func(element)] = element
+
+
+@toolz.curry
+def _accumulate(
+    element: Any, is_seen: Callable, better_node: Callable, keep_best: bool
+):
+    return not is_seen(element) or (keep_best and better_node(element))
+
+
 @toolz.curry
 def graph_traverse(
-    source: Any, get_neighbors: Callable, key: Callable = toolz.identity
+    source: Any,
+    get_neighbors: Callable,
+    key: Callable = toolz.identity,
+    keep_best: bool = False,
 ) -> Iterable:
-    yield from graph_traverse_many([source], get_neighbors=get_neighbors, key=key)
+    seen_dict: Dict = {}
+    remember = _save_to_dict(dictionary=seen_dict, func=key)
```

Comment:
assigning these to variables hinders readability (the reader needs to search for the usage and must read all lines in between), better to inline them in the function below

Reply:
So I can keep the same signature and parameters for `graph_traverse_many` and I create all of these functions inside it